### PR TITLE
Mock the new module API on <38 and ignore second argument.

### PR DESCRIPTION
**Pull request by Carreau**
As ast.Module is used only in interactiveshell the mock is only a small
lambda function but would need to grow into a proper shim if used more
widely.

### CODE REVIEW

**Comment by Carreau** [comment]
@njsmith mind giving it a try ? I'm going to guess ast.changes will will likely have ripple in Jedi as well.

**Comment by tacaswell** [comment]
What behavior should this effect?

**Comment by tacaswell** [comment]
Via a terrible shell script (https://gist.github.com/tacaswell/a701178de7fcc0c213c0adb0abd57dff ).  Basically build cpython, create a venv, pip install into it.

Ah, I now see the issue, kinda hard to miss.... :sheep:

**Comment by tacaswell** [comment]
And yes, this branch fixes it.

**Comment by njsmith** [comment]
Change looks good to me. Should you be adding `3.8-dev` to your Travis matrix to catch things like this in the future?

**Comment by Carreau** [comment]
I've added 38-dev, but as we are failing on nightly with the ast transformers, i'm going to assume it will fail as well, and someone need to investigate why some transformations are not being applied.

**Comment by Carreau** [comment]
and so apparently `Num` has been renamed to `Constant`  in the ast ?

**Comment by Carreau** [comment]
Ok, 3 more test fixed. @takluyver, I might need help on some of these errors.

**Comment by Carreau** [comment]
It's unclear to me what's the difference between `node.n` and `node.value` are they alias to each other now ? If so should one be deprecated ? 

It also seem that both `Num`, and `Str` are now folded into the `Constant` ast node.

**Comment by Carreau** [comment]
@asmeurer , @williamstein don't you have ast transformers ? If so that will affect you.

**Comment by Carreau** [comment]
Ok, test are passing. I'd love for someone to make sure the code make sens, I wrote that by chunks of 5 min here and there so it might not be super coherent.

**Comment by asmeurer** [comment]
I believe we are still using custom transformers based on tokenize. https://github.com/sympy/sympy/blob/master/sympy/interactive/session.py

File: IPython/core/tests/test_interactiveshell.py [hunk 1]
```diff
@@ -609,10 +609,17 @@ def test_extraneous_loads(self):
 
 class Negator(ast.NodeTransformer):
     """Negates all number literals in an AST."""
+
     def visit_Num(self, node):
         node.n = -node.n
         return node
 
+    if sys.version_info > (3,8):
```
[the remark this answers is not in the document]
Comment (Carreau) [reply IPython/core/tests/test_interactiveshell.py]: Thanks for the review, I added this to every relevant point.

File: IPython/core/tests/test_ultratb.py [hunk 2]
```diff
@@ -379,10 +379,12 @@ def eggs(f, g, z=globals()):
         handler(*sys.exc_info())
     buff.write('')
 
+from IPython.testing.decorators import skipif
 
 class TokenizeFailureTest(unittest.TestCase):
     """Tests related to https://github.com/ipython/ipython/issues/6864."""
 
+    @skipif(sys.version_info > (3,8))
```
[the remark this answers is not in the document]
Comment (Carreau) [reply IPython/core/tests/test_ultratb.py]: This was apparently testing we properly handle a bug in tokenizer that was fixed in 3.8, so the test is not relevant anymore.

**Comment by Carreau** [comment]
Hi there, You seem to run IPython on Python 3.8; do you mind trying #11713 that tried to make use of the new 3.8b1 (not out yet) top-level await ?

**Pull request merged**